File: frontend/src/types/Cloudbet.ts
export type CloudbetTeam = {
    name: string
    key: string
}
export type CloudbetSelection = {
    outcome: string
    params: string
    price: number
    minStake: number
    maxStake: number
    side: 'BACK' | 'LAY'
    status: 'SELECTION_DISABLED' | 'SELECTION_ENABLED'
    
}
export type CloudbetMarket = {
    submarkets: {
        [key: string]: {
            selections: CloudbetSelection[]
            sequence: string
        }
    }
}
export type CloudbetMarkets = {
    [key: string]: CloudbetMarket
}
export type CloudbetEvent = {
    away: CloudbetTeam
    home: CloudbetTeam
    markets: CloudbetMarkets
    name: string
    status: string
    cutoffTime: string
    key: string
}
export type CloudbetCompetition = {
    category: {
        name: string
        key: string
    }
    events: CloudbetEvent[]
    name: string
    key: string
}

export interface CloudbetOddsResponse {
    competitions: CloudbetCompetition[]
}

export interface CloudbetCompetitionsResponse {
    categories: {
        competitions: {
            name: string
            key: string
            eventCount: number
        }[]
        key: string
        name: string
    }[]
    key: string
    name: string
}

File: backend/models/cloudbet.py
from enum import Enum
from typing import List, Dict, Union, Optional
from pydantic import BaseModel, constr, RootModel


# Define your enums
class SideEnum(str, Enum):
    BACK = 'BACK'
    LAY = 'LAY'

class StatusEnum(str, Enum):
    SELECTION_DISABLED = 'SELECTION_DISABLED'
    SELECTION_ENABLED = 'SELECTION_ENABLED'
    
    
class CloudbetTeam(BaseModel):
    name: str
    key: str
    abbreviation: str
    nationality: str

class CloudbetSelection(BaseModel):
    outcome: str
    params: str
    price: float
    minStake: float
    maxStake: float
    side: SideEnum
    status: StatusEnum

class CloudbetSubmarket(BaseModel):
    selections: List[CloudbetSelection]
    sequence: str

class CloudbetMarket(BaseModel):
    submarkets: Dict[str, CloudbetSubmarket]
    
class CloudbetMarkets(RootModel):
    root: Dict[str, CloudbetMarket]

class CloudbetCompetitionWithCategory(BaseModel):
    description: Optional[str] = None
    key: str
    name: str
    
class CloudbetEvent(BaseModel):
    away: CloudbetTeam
    home: CloudbetTeam
    markets: CloudbetMarkets
    name: str
    status: str
    cutoffTime: str
    key: str
    id: int
    competition: Optional[CloudbetCompetitionWithCategory] = None

class CloudbetCompetition(BaseModel):
    events: List[CloudbetEvent]
    name: str
    key: str


